Handlers/ProxyUserIpAuditTrailEventHandler.cs

```diff
@@ -0,0 +1,29 @@
+using Microsoft.AspNetCore.Http;
+using OrchardCore.AuditTrail.Services;
+using OrchardCore.AuditTrail.Services.Models;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Lombiq.AuditTrailExtensions.Handlers
+{
+    public class ProxyUserIpAuditTrailEventHandler : AuditTrailEventHandlerBase
```

Comment:
E helyett miért nem a Reverse Proxy Configuration feature-t kapcsolod be és configolod recipe-ből?

Reply:
A feature bekapcsolása megvolt, úgy sem volt jó. Nem tudtam, hogy recipeből bármit is lehetne állítani. Erről van valami dokumentáció, hogy miként kellene használni? Kódból nekem nem derült ki.

Reply:
Ja igazából ugyan azt lehetne elérni vele, mintha settingsből enable-ket bekapcsolom, nem?

Reply:
Igen. Most is vannak már site settingseink a recipe-kben. De a bekapcsolást és configolást az After resolve-ba dokumentáld, hogy deploy után is legyen megcsinálva (egy fokkal szebb lenne egy adminról lefuttatott recipe, de ezért a három checkboxért nem látom értelmét).

Reply:
Őm, szóval azt mondod, ha a feature enabled és három checkbox be van kapcsolva, akkor működnie kell? Mert tested és sajnos nem, és ezért lett ez.

Reply:
Igen, annak mennie kellene (legalábbis az aktuális OC verzióval, lehet amit mi használunk, abban valami bug van).

Ezzel a megoldással az a probléma, hogy bár konkrétan ezt az esetet megoldja, de a `ClientIpAddress` lekérdezése mindenhol máshol is hibás eredményt fog adni. A Reverse Proxy Configuration modul által használt .NET Core-os megoldás viszont ezt általánosan megoldja.

Reply:
Sad.

Reply:
Akkor utánajárok, hogy miként lehet megheggeszteni, ha meglehet.

Reply:
A Helpful Librariesben most adtam hozzá a `UseForwardedHeadersForCloudflareAndAzure()` metódust. Használd azt, meg fogja oldani az egészet. Figyelj, hogy `dev`-ből mergelni kell, de az már 1.0-ás branch.

Reply:
Köszi! Múltkor rájöttem amúgy én is, hogy valszeg elég lett volna beállítani a ForwardLimit-et a default 1-ről 2-re vagy 3-ra. 
De ezen kívül azt gondoltam, hogy a known networksbe is bele kellene pl a cloudfare IP tartományait rakni, így arra elkezdtem egy admin editort összerakni, hogy ott a known networkshöz hozzá lehessen addolni egyszerűen. De nem jött az össze, valamiért mindig kétszer hívta meg a displaydriverben az `Edit()`-et és ezért törlődött az amúgy elmentett adat (furi). Szóval ez maradt így, mert nagyobb prio issuet kellett csinálnom.

Reply:
Viszont ami nem tiszta, hogy a Helpful Librariesben van egy orchard-core-preview branch illetve egy dev branch, amiben most ez `UseForwardedHeadersForCloudflareAndAzure()` megtalálható . Eddig emlékeim szerint mindig a orchard-core-preview branchet használtuk. Szóval ez most hogy van?

Reply:
Igen, ahogy írtam, mergelned kell a preview-ba, hogy használhasd. Pontosabban valszeg egyszerűbb az utóbbi deves commitokat cherry pickelni (azaz átmásolni a preview branchre a git cherry pick commandokkal).

Reply:
Jaa, hát a dev-ből mergelni kellel nem jött át, hogy te erre gondolsz és nem a Finitive repojának a dev branchét kell mergelni.

Reply:
Ezt a PR-t zárom, mert nem kell.